cmds/cpud/serve_test.go

```diff
@@ -34,14 +37,20 @@ func TestListen(t *testing.T) {
 		{"tcp", "17010"},
 		{"tcp4", "17010"},
 		{"tcp6", "17010"},
-		// Damn. Can't easily test this b/c environments vary so much.
-		//{"vsock", "17010"},
+		{"vsock", "17010"},
 		{"unix", "@foobar"},
 	}
 
 	for _, tt := range oktests {
 		ln, err := listen(tt.network, tt.port)
 		if err != nil {
+			var sys_err *os.SyscallError
+			if errors.As(err, &sys_err) {
```

Comment:
nit: I think you can get rid of the switch and case with just:
```
// Ignore failures caused by lack of ipv4 or vsock.
if errors.As(err, &sys_err) && sys_err.Err == syscall.EAFNOSUPPORT {
  continue
}
```

Reply:
done.  sorry i missed this earlier.  i got a github email about lencerf's comment, but not yours.